rfcs/text/0000-create-file-macho.md

```diff
@@ -0,0 +1,146 @@
+# 0000: Create the Mach-O sub-field of the File fieldset
+
+- Stage: **1 (draft)**
+- Date: **TBD**
+
+Create the Mach Object (Mach-O) sub-field, of the `file` or `process` top-level fieldsets. This document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+**Stage 0**
+
+This RFC is to create the Mach-O sub-field within the `file.` fieldset. This will include 35 sub-fields.
+
+|   Name                                     |   Type     |   Description                                                               |
+|--------------------------------------------|------------|-----------------------------------------------------------------------------|
+|   macho.cpu                          |   object   |   CPU information for the file.                                             |
+|   macho.cpu.architecture             |   keyword  |   CPU architecture target for the file.                                     |
```

Comment:
@andrewstucki I did some cross-checking between the proposed field list here and your work in https://github.com/elastic/beats/pull/24195. I pulled those mappings for `macho` into [a gist](https://gist.github.com/ebeahan/41507b47bcadff43f866350634ffa9ae) for easier reference.

100% alignment isn't necessary at all at this point. However, there are enough differences between that approach and what's captured here I think it's worth raising.

@peasead @dcode @devonakerr

Reply:
I have since updated this PR with a proposed layout that aligns sections and segments with ELF and PE (which doesn't have segments).

Reply:
Thanks, @dcode. 

There's still some large differences between the implementation in elastic/beats#24195 and what's proposed here.

Again, not something we need to completely resolve now. At a minimum, let's capture the difference in approaches as a `Concern` to make sure we revisit the topic in the next stage's PR.

Reply:
@andrewstucki the strategy I'm suggesting here is splitting out fat binary components that can be unified by the same file metadata (namely hashes or other source unique id's). That will allow analyzing all binaries by their sections, segments, symbols, etc in a seamless way, regardless of platform or if their multi-platform. Do you think that can work for your implementation?

Reply:
Just pining to see if there was a resolution here.

Reply:
Just as a use-case example: let's say we're triggering an alert on a malicious binary that has been compiled for both the new m1 as well as older, intel-based chipsets and we want to add in object-level data about the file. Would we emit two separate alerts, one for each architecture? Seems kind of odd.